### PR TITLE
GitHub templates: Use GitHub issue forms

### PR DESCRIPTION
**Pull request by travier**
See https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema

Those are still in Beta. Not sure if we want to convert all templates yet.

You can try it on my fork https://github.com/travier/fedora-coreos-tracker/issues/new?assignees=&labels=kind%2Fbug&template=bug-report.yml.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    validations:
+      required: true
+
+  - type: textarea
+    id: bug-ignition
+    attributes:
+      label: Ignition config
+      description: Please attach the Butane config or Ignition config used to provision your system. Be sure to sanitize any private data. If not using Butane to generate your Ignition config, does the Ignition config pass validation using [ignition-validate](https://coreos.github.io/ignition/getting-started/#config-validation)?
+    validations:
+      required: true
```
[the remark this answers is not in the document]
Comment (travier) [reply .github/ISSUE_TEMPLATE/bug-report.yml:62]: Sure

[the remark this answers is not in the document]
Comment (travier) [reply .github/ISSUE_TEMPLATE/bug-report.yml:62]: I think you may also answer N/A to skip it.

**Comment by travier** [comment]
I've updated all templates. It's mostly copy/paste with some minor additions

**Comment by travier** [comment]
Try it at https://github.com/travier/fedora-coreos-tracker/issues/new/choose

**Review by jlebon — APPROVED** [review]
Cool! This looks sane to me. The only one I'm not sure about is the new platform one where having really long bolded questions looks a bit odd. But not a big deal.

File: .github/ISSUE_TEMPLATE/new-package.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
+name: Request a new package
+description: Ask for a new package to be added to Fedora CoreOS
+title: "New Package Request: <package name>"
+labels: ["kind/enhancement"]
+assignees: []
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Please try to answer the following questions about the package you are requesting
```
Comment (bgilbert) [inline .github/ISSUE_TEMPLATE/new-package.yml:10]:
```suggestion
        Please try to answer the following questions about the package you are requesting.
```

File: .github/ISSUE_TEMPLATE/new-package.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      label: What, if any, are the additional dependencies on the package? (i.e. does it pull in Python, Perl, etc)
+      description: Paste here the output of `rpm-ostree install --dry-run <package>` from a fresh Fedora CoreOS node.
+    validations:
+      required: true
+
+  - type: textarea
+    id: newpackage-size
+    attributes:
+      label: What is the size of the package and its dependencies?
+      description: Paste here the output of `rpm -qi <pacakge>` for each package mentionned above.
```
Comment (bgilbert) [inline .github/ISSUE_TEMPLATE/new-package.yml:24]:
```suggestion
      description: Paste here the output of `rpm -qi <package>` for each package mentioned above.
```

File: .github/ISSUE_TEMPLATE/new-package.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    attributes:
+      label: Can the tool(s) provided by the package be helpful in debugging networking issues?
+    validations:
+      required: true
+
+  - type: textarea
+    id: newpackage-day2
+    attributes:
+      label: Is it possible to layer the package onto the base OS as a day 2 operation?  Explain why or why not.
+      description: Can the package be installed on first boot on later with `rpm-ostree install <package>`.
```
Comment (bgilbert) [inline .github/ISSUE_TEMPLATE/new-package.yml:60]:
```suggestion
      description: Can the package be installed on first boot or later with `rpm-ostree install <package>`?
```

File: .github/ISSUE_TEMPLATE/new-platform.yml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+      label: Does the platform have an agent that runs inside the instance?  Is it required?  What does it do?  What language is it implemented in, and where is the source code repository?
+    validations:
+      required: true
+
+  - type: textarea
+    id: newplatform-image-upload
+    attributes:
+      label: How are VM images uploaded to the platform and published to other users?  Is there an API?  What disk image format is expected?
+    validations:
+      required: true
```
Comment (bgilbert) [inline .github/ISSUE_TEMPLATE/new-platform.yml:80]: Unlike the new-package form, where the submitter can reasonably be expected to know the answers, this form requires a lot of information that will probably take some time to collect.  Should we make a bunch of these `required: false`?

**Review by bgilbert — COMMENTED** [review]
This looks great!  :+1:

File: .github/ISSUE_TEMPLATE/new-platform.md [hunk 6]
```diff
@@ -1,32 +0,0 @@
...
-about: Ask for Fedora CoreOS to support a new cloud environment
-title: 'Platform Request: <platform name>'
-labels: 'area/platforms, kind/enhancement'
-assignees: ''
-
----
-
-In order to implement support for a new cloud platform in Fedora CoreOS, we need to know several things about the platform.  Please try to answer as many questions as you can.
-
-- [ ] Why is the platform important?  Who uses it?
```
[the remark this answers is not in the document]
Comment (travier) [reply .github/ISSUE_TEMPLATE/new-platform.md:12]: Unfortunately I don't have a better idea for this one

**Review by bgilbert — APPROVED** [review]

**Review by cgwalters — APPROVED** [review]

**Comment by zaneb** [comment]
The link in the README under https://github.com/coreos/fedora-coreos-tracker#adding-packages-to-fedora-coreos needs to be updated.

**Comment by travier** [comment]
Thanks: https://github.com/coreos/fedora-coreos-tracker/pull/1461

**Pull request merged**